docs/submitting-applications.md

```diff
@@ -48,6 +48,44 @@ Some of the commonly used options are:
 * `application-jar`: Path to a bundled jar including your application and all dependencies. The URL must be globally visible inside of your cluster, for instance, an `hdfs://` path or a `file://` path that is present on all nodes.
 * `application-arguments`: Arguments passed to the main method of your main class, if any
 
+Alternatively, for submitting on yarn, 
+
+{% highlight bash %}
+./bin/spark-submit \
+  --class <main-class>
+  --master <yarn-deploy-mode>
+  --conf <key>=<value> \
+  ... # other options
+  <application-jar> \
+  [application-arguments] 
+{% endhighlight %}
+
+* `--master`: The --master parameter is either `yarn-client` or `yarn-cluster`. Defaults to `yarn-client`
```

Comment:
I don't think it makes sense to promote both ways. I also think both have to be retained for compatibility. I'm still not clear what people think the usual way to do it is; I've always seen and used `yarn-client` for example. In the end, after that's established, the docs should feature the preferred way, and mention the other in passing.